android-kit-base/src/main/java/com/mparticle/kits/KitManagerImpl.java

```diff
@@ -1553,20 +1563,8 @@ else if (hashedEmailMismatch && existingHashedEmail != null) {
                 if (emailMismatch) {
                     identityBuilder.email(email);
                 }
-                if (hashedEmailMismatch && kitConfiguration != null) {
-                    MParticle.IdentityType type = null;
-                    try {
-                        String identityTypeStr = kitConfiguration.getHashedEmailUserIdentityType();
-                        if (identityTypeStr != null && !identityTypeStr.equalsIgnoreCase("UNASSIGNED")) {
-                            type = MParticle.IdentityType.valueOf(identityTypeStr);
-                        }
-                    } catch (IllegalArgumentException e) {
-                        type = MParticle.IdentityType.Other; // fallback if invalid
-                    }
-
-                    if (type != null) {
-                        identityBuilder.userIdentity(type, hashedEmail);
-                    }
+                if (hashedEmailMismatch && selectedIdentityType != null) {
```

Comment:
Pretty sure `selectedIdentityType != null` can be removed.  If i am misunderstanding, please let me know